### PR TITLE
Chore: finish networks refactoring

### LINKED ISSUE #13839: Refactor and type network config
There is file `networksConfig.ts`

In this file, there is `networks` variable declaring all networks available in Suite and there is also `networksCompatibility` which is a transformation of `networks` to format used in the past.

**Issues:**
- `networks` variable is not typed, the type is generated from the structure of file
  - it would be best to type it so that it is easier to work with it
- `networksCompatibility ` is dynamic and it is almost a blackbox how to work with it. It would be great to finally get rid of it in behalf of `networks`

- together with those changes we can maybe change (as a bonus, not really needed but would be nice)
  - `explorer` can have `base` url and then just `path` for `tx, account, address`,...
  - `default` account type which is now on root of each network could be moved to `accountTypes` under `default` type
  - `customBackends` can be called just `backendTypes` or `backends`
  - I am not sure if we really need `support`, it is also defined in `coins.json`
    - cardano: old fw version of T is deprecated anyway so we can just bump supported version in there maybe
    - solana is 1:1
    - probably we need it but extended for seed type, see Optimisation #13738

**Why:**
- cleaner codebase
- easier work
- easier to implement - different `networkSymbol` from `networkCurrency` required for L2
  - `ln`: `BTC`
  - `op`: `ETH`
  - `base`: `ETH`
  - ...

### PR DESCRIPTION
**Pull request by Lemonexe**
Finish refactoring of old `networksCompatible` to `networks` :tada:

## Description

- 342638d3792077b5fb5031a195442433ad320c9e the heavy lifting :muscle: These changes had to be done together, as they are tightly interlinked:
  - refactor `selectedAccount` state
  - refactor discovery (a lot of logic changed!) 
  - all the easy stuff related to either discovery || selected account
- 3332ef75f45f96c05155d19e033170e3c0acd4c5 nuke legacy code :broom: :bucket: :radioactive:
- d0a25d92d563e08cf2c462d27c8705bffcd9851c incidental fix of pre-existing discovery bug :bug: with `tada` :tada: , discovery would seemingly never finish

## Related Issue

Resolve #13839 , finally :relaxed: :tada:

## Screenshots:

I tested in app that the discovery works as before: all accounts are discovered & unfinished/in progress is indicated correctly. But to be **really** sure, I console logged [`loaded` and `total` at `calculateProgress#L150`](https://github.com/trezor/trezor-suite/blob/d0a25d92d563e08cf2c462d27c8705bffcd9851c/suite-common/wallet-core/src/discovery/discoveryThunks.ts#L150)
- I activated every coin except `tada` on T2T1 w/ academic seed, and this is how the discovery progressed on this branch vs. develop: [discovery.test.txt](https://github.com/user-attachments/files/17060772/discovery.test.txt)
 (it's the same, both converged on `65 / 65` :heavy_check_mark: )
- I verified the same with another wallet with even more accounts, both converged on `92 / 92`
- before :tada: :bug: was solved, it'd converge on for example `16 / 36`, indicating unfinished discovery forever:
tada bug

### CODE REVIEW

File: suite-common/wallet-core/src/discovery/discoveryThunks.ts [hunk 1]
```diff
@@ -74,44 +73,32 @@ export const filterUnavailableAccountTypes = (
...
+        ).length;
+
+        // numbers of accounts of givent types
+        const { numberOfNonCardano, numberOfCardano } = networksToCount.reduce(
+            (acc, network) => {
+                const { symbol } = network;
+
+                // increment the appropriate counter based on the network symbol
+                const incrementAccumulator = () => {
+                    if (symbol === 'ada' || symbol === 'tada') acc.numberOfCardano += 1;
```
Comment (Lemonexe) [inline suite-common/wallet-core/src/discovery/discoveryThunks.ts:102]: Note: at first I thought just including `'tada'` here would solve the :tada: :bug:,
but I also had to do the "1 or 2 thing" with `numberOfCardanoCoins` instead of always 1.

Otherwise, with `ada` & `tada` at the same time, discovery was wrongly indicated as finished, with debug showing `16 / 6` as loaded / total :scream:
I would have to activate some more coins to "manually" start discovery of `ada` or `tada`..

File: suite-common/test-utils/src/mocks.ts [hunk 2]
```diff
@@ -560,8 +560,7 @@ const intlMock = {
     formatMessage: (s: any) => s.defaultMessage,
 };
 
-const mockedBlockchainNetworks = networksCompatibility.reduce((result, network) => {
-    if (network.accountType) return result;
```
Comment (Lemonexe) [inline suite-common/test-utils/src/mocks.ts:562]: this line means that when using `networksCompatibility`, only `'normal'` accounts were processed, which makes the refactoring really easy :relaxed: Analogical diff in several other files below.

File: suite-common/wallet-config/src/networksConfig.ts [hunk 3]
```diff
@@ -641,44 +633,3 @@ export const networks = {
...
-    ({ symbol, accountTypes, ...network }) => [
-        { symbol, ...network },
-        ...Object.entries(accountTypes).map(([accountType, networkOverride]) => ({
-            symbol,
-            accountType,
-            ...network,
-            ...networkOverride,
-        })),
-    ],
-);
```
Comment (Lemonexe) [inline suite-common/wallet-config/src/networksConfig.ts:684]: farewell :wave:

File: suite-common/wallet-core/src/accounts/accountsThunks.ts [hunk 4]
```diff
@@ -33,7 +33,7 @@ export const disableAccountsThunk = createThunk(
         const accounts = selectAccounts(getState());
         const enabledNetworks = selectEnabledNetworks(getState());
         // find disabled networks
-        const disabledNetworks = networksCompatibility
+        const disabledNetworks = networksCollection
             .filter(n => !enabledNetworks.includes(n.symbol) || n.isHidden)
             .map(n => n.symbol);
```
Comment (Lemonexe) [inline suite-common/wallet-core/src/accounts/accountsThunks.ts:38]: :information_source: this is different! With `networksCompatibility`, the array of symbols is not unique, so e.g. `'btc'` is duplicated foreach account. Now with `networksCollection` it is unique.
BUt in this case it's ok, see L41, `disabledNetworks` is only used to test `.includes`

File: suite-common/wallet-core/src/blockchain/blockchainThunks.ts [hunk 5]
```diff
@@ -61,7 +61,7 @@ export const preloadFeeInfoThunk = createThunk(
     `${BLOCKCHAIN_MODULE_PREFIX}/preloadFeeInfoThunk`,
     async (_, { dispatch }) => {
         // Fetch default fee levels
-        const networks = networksCompatibility.filter(n => !n.isHidden && !n.accountType);
```
Comment (Lemonexe) [inline suite-common/wallet-core/src/blockchain/blockchainThunks.ts:65]: filters only `'normal'` accounts, so it's the same :heavy_check_mark:

File: suite-common/wallet-core/src/discovery/discoveryThunks.ts [hunk 6]
```diff
@@ -74,44 +73,32 @@ export const filterUnavailableAccountTypes = (
...
     (discovery: Discovery) =>
     (_dispatch: any, getState: any): PartialDiscovery => {
-        const { numberOfNonCardano, numberOfCardano } = discovery.networks.reduce(
-            (acc, symbol) => {
-                if (symbol === 'ada') {
-                    acc.numberOfCardano += 1;
-                } else {
-                    acc.numberOfNonCardano += 1;
-                }
+        const device = selectDevice(getState());
```
Comment (Lemonexe) [inline suite-common/wallet-core/src/discovery/discoveryThunks.ts:87]: what's going on here? See [L741, L748](https://github.com/trezor/trezor-suite/blob/ced3f2a8a54c2636010c61e1f11be1bce03c692c/suite-common/wallet-core/src/discovery/discoveryThunks.ts#L741-L748) where it's first called.

**Previously**, `networksSymbols` from `filterUnavailableNetworksCompatible` was non-unique, [like in comment above](https://github.com/trezor/trezor-suite/pull/14398/files/d8e95808ac13344515846ca1d08255ad06b957a4..ced3f2a8a54c2636010c61e1f11be1bce03c692c#r1764541695), except here **it actually mattered**, and imho the previous implementation was kinda occult.
In `calculateProgress`, we need the _number_ of loaded accounts, so the number of duplicated `discovery.networks` symbols was counted.

**Now**, `discovery.networks` is unique, and it is responsibility of `calculateProgress` to iterate through accounts.

:bulb:  Note how `filterUnavailableNetworksCompatible` is split between `filterUnavailableNetworks` and `filterUnavailableAccountTypes`.

File: suite-common/wallet-core/src/discovery/discoveryThunks.ts [hunk 7]
```diff
@@ -279,58 +265,64 @@ export const getBundleThunk = createThunk(
...
-                    networkType: configNetwork.networkType,
+                    networkType,
                     derivationType: getDerivationType(accountType),
                     suppressBackupWarning: true,
                 });
             }
-        });
+        };
+
+        // foreach network, start discovery foreach normalized account (normal as well as all alternative accounts)
```
Comment (Lemonexe) [inline suite-common/wallet-core/src/discovery/discoveryThunks.ts:320]: "normalized account" is a very helpful helper introduced in [#14281 here](https://github.com/trezor/trezor-suite/commit/8f0270267d714f747542facdee446861ef20ed69), and used in `filterUnavailableAccountTypes`.

**TL;DR:** It's kinda like `networksCompatibility` but **only** for accounts, which actually does make sense!
It gets all accounts of network, but 1) synthesizes a `'normal'` account from the network properties, and 2) for the alternative accs it backfills missing properties from normal.

:arrow_right: never again u have to do stuff like `account.bip43Path ?? network.bip43Path`, because it's very easy to do a mistake there :hear_no_evil:
You just iterate over all accounts, and the correct properties are guaranteed to be there :ok_hand:

File: suite-common/wallet-core/src/discovery/discoveryThunks.ts [hunk 8]
```diff
@@ -709,17 +701,23 @@ export const createDiscoveryThunk = createThunk(
...
+        const maxTotalAccounts = LIMIT * availableConfigAccounts;
 
         dispatch(
             createDiscovery({
                 deviceState,
                 authConfirm: !device.useEmptyPassphrase,
                 index: 0,
                 status: DiscoveryStatus.IDLE,
-                total: LIMIT * networksSymbols.length,
+                total: maxTotalAccounts,
```
Comment (Lemonexe) [inline suite-common/wallet-core/src/discovery/discoveryThunks.ts:720]: I suspect this is not used anywhere? But I'm reaaally not sure. I updated it anyway.

File: suite-native/module-send/src/screens/SendOutputsScreen.tsx [hunk 9]
```diff
@@ -163,9 +163,9 @@ export const SendOutputsScreen = ({
         // Keyboard has to be dismissed here before navigating, so it's animation is not interfering with the animations on the FeesScreen.
         Keyboard.dismiss();
 
-        const network = getNetworkCompatible(account?.symbol);
+        if (!account) return;
 
-        if (!network) return;
+        const network = networks[account.symbol];
```
Comment (Lemonexe) [inline suite-native/module-send/src/screens/SendOutputsScreen.tsx]: when `!!account`, `network` is now guaranteed :heavy_check_mark:

File: suite-common/wallet-core/src/discovery/discoveryThunks.ts (hunk 7 above)
Comment (tomasklim) [reply suite-common/wallet-core/src/discovery/discoveryThunks.ts:320]: Thank you! and for tests too!

File: suite-common/wallet-core/src/discovery/discoveryThunks.ts (hunk 1 above)
Comment (tomasklim) [reply suite-common/wallet-core/src/discovery/discoveryThunks.ts:102]: Thank you! (Maybe I am just blind but it would be worth it to add tests for that?)

**Review by tomasklim — APPROVED** [review]
🎉🎊🥳🎈🎶🍾🍻🍹🍸🍷🥂🍺🎂🎇🎆💃🕺🎧🎤🎵🎺🎸🎻🥁🎡🎠🎢🚀🪅🍕🍔🌭🍿🍰🧁🍨🍩🍭🍬🍫🪩🎯🕹️🎲💥🔥🌟✨💫🌈👯‍♂️👯‍♀️👑👒🥳

File: suite-common/wallet-core/src/discovery/discoveryThunks.ts (hunk 1 above)
Comment (Lemonexe) [reply suite-common/wallet-core/src/discovery/discoveryThunks.ts:102]: Good idea. I'd put it into followup work, so that we can merge this, and not accumulate conflicts - after all, there were no tests previously.
EDIT: added to the [appropriate issue](https://github.com/trezor/trezor-suite/issues/14444)
